backend/user_service/controllers/userController.js

```diff
@@ -3,8 +3,10 @@ const admin = require('firebase-admin');
 
 // Function to create a new user and add them to the Firestore collection
 const addToUserCollection = async (req, res) => {
-    const { uid, email, displayName } = req.body;
-
+    const { uid, email, displayName, username } = req.body;
+    if (!/^[a-zA-Z0-9]+$/.test(username)) {
+        return res.status(400).json({ success: false, message: "Username must be alphanumeric only." });
```

Comment:
TBH feel like we should have had a errors.ts file for all our errors for consistency sake but i guess for now this is fine too. Realised a lil too late

Reply:
could make an issue to KIV?

Reply:
HAHA it means keep in view :>